### PR TITLE
Nc news 11 get articlesbyquery

### PR DESCRIPTION
**Pull request by rsdecoder**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -105,7 +106,42 @@ describe("GET", () => {
...
+        })
+      })
+    })
+    test("STATUS 404, should return an appropriate messasge if article does not exist with the matched query", () => {
+      return request(app)
+      .get("/api/articles?topic=ragini")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe('article does not exist for matched query');
+      })
```
Comment (nc-rayhaan) [inline __tests__/app.test.js:140]: I feel the error message doesn't match here. It should be something like the topic doesn't exist

File: __tests__/app.test.js [hunk 2]
```diff
@@ -105,7 +106,42 @@ describe("GET", () => {
...
+    })
+
```
Comment (nc-rayhaan) [inline __tests__/app.test.js:142]: Missing a test here. What if the topic does exist but has no articles? For example `topic=paper`?

File: models/articles.models.js [hunk 3]
```diff
@@ -11,8 +11,18 @@ exports.selectArticlesById = (article_id) => {
...
+  
+  if(topic) {
+    return db.query(`SELECT * FROM articles WHERE topic = $1`, [topic])
+    .then((result) => {
+      if(result.rows.length === 0){
+        return Promise.reject({status: 404, msg: 'article does not exist for matched query'})
+      }
+      return result.rows;
+    })
+  }
```
Comment (nc-rayhaan) [inline models/articles.models.js:24]: Wit this query, you are asking SQL for all info of articles. This also includes body which shouldn't be included. Also there is no comment count for these. Please have a go at refactoring this model to make use of just one db.query. The notes on complex queries will really help with this 😄

**Review by nc-rayhaan — COMMENTED** [review]
Almost there with this one, Ragini. Left some comments for you. Please fix and resubmit once done 🙂

File: models/articles.models.js (hunk 3 above)
Comment (rsdecoder) [reply models/articles.models.js:24]: Will do. I was wondering how I would do that!

**Pull request merged**